pymeasure/instruments/hp/hp437b.py

```diff
@@ -0,0 +1,949 @@
+    33: "Both front and rear sensor inputs, have sensors connected (Option 002 or Option 003 only)",
+    50: "Entered cal factor is out of range",
+    51: "Entered offset is out of range",
+    52: "Entered range number is out of range",
+    54: "Entered recall register number is out of range",
+    55: "Entered storage register number is out of range",
+    56: "Entered reference cal factor is out of range",
+    57: "RAM ID check failure",
+    61: "Stack RAM failure",
+    62: "ROM checksum failure",
+    64: "RAM failure",
+    65: "Analog I/O PIA Failure",
+    66: "Keyboard and Display PIA Failure",
+    67: "Analog-to-Digital converter Failure",
+    68: "HP-IB failure",
+    69: "Timer failure",
+    70: "Keyboard/Display controller failure",
+    71: "Keyboard data failure",
+    72: "Data line to A3U26 is open.",
+    73: "Keyboard/Display controller self-test failure",
+    74: "Display not responding",
+    75: "Digital failure",
+}
+
+
+class StatusMessage:
+    MeasurementErrorCode = (0, 1)
+    EntryErrorCode = (2, 2)
+    OperatingMode = (4, 2)
+    AutomaticRangeStatus = (6, 1)
+    Range = (7, 1)
+    # 8, 9 unused
+    AutoFilterStatus = (10, 1)
+    Filter = (11, 1)
+    # 12, 13 unused
+    LinearLogStatus = (14, 1)
+    # A
+    PowerRefStatus = (16, 1)
+    RelativeModeStatus = (17, 1)
+    TriggerMode = (18, 1)
+    GroupTriggerMode = (19, 1)
+    LimitsCheckingStatus = (20, 1)
+    LimitsStatus = (21, 1)
+    # 22 unused
+    OffsetStatus = (23, 1)
+    DutyCycleStatus = (24, 1)
+    MeasurementUnits = (25, 1)
+
+
+class HP437B(Instrument):
+    """Represents the HP437B Power Meters.
+
+    .. note::
+        Most command descriptions are taken from the document:
+        'Operating Manual 437B Power Meter'
+    """
+
+    def __init__(self, adapter, name="Hewlett-Packard HP437B", **kwargs):
+        super().__init__(
+            adapter,
+            name,
+            includeSCPI=False,
+            send_end=True,
+            **kwargs,
+        )
+
+    def __getstatus(status_type: StatusMessage, modifier=lambda v: v):
+        start_index, stop_offset = status_type
+        return lambda v: modifier(int(v[start_index:start_index + stop_offset]))
+
+    def check_errors(self):
+        errors = []
+        while True:
+            err = self.values("ERR?")
+            # exclude upper limit and lower limit hit from real errors
+            if int(err[0]) != 0 and int(err[0]) != 21 and int(err[0]) != 23:
+                log.error(f"{self.name}: {err[0]}, {Errors[err[0]]}")
+                errors.append(err)
+            else:
+                break
+        return errors
+
+    event_status = Instrument.measurement(
+        "*ESR?",
+        """
+        Get the status byte and Master Summary Status bit.
+
+        .. code-block:: python
+
+            print(instr.request_service_conditions)
+            StatusRegister.PowerOn|CommandError
+        """,
+        get_process=lambda v: EventStatusRegister(int(v))
```

Comment:
Just that you know: you can use the `cast` attribute to make the integer conversion:
`cast=int`.